composer.json

```diff
@@ -46,6 +46,7 @@
         "sylius/grid": "self.version"
     },
     "conflict": {
+        "pagerfanta/pagerfanta": "^3.0",
```

Comment:
You need to conflict the individual packages as well, after the Pagerfanta 2.4 releases there is no guarantee that downstream users are installing the `pagerfanta/pagerfanta` mono-package.

Based on the issue report, `pagerfanta/doctrine-dbal-adapter` needs to be added at a minimum.

You'll also need a conflict blocking versions older than 2.4 as that's when the newer classes were introduced.

Reply:
@mbabker Oh you're right.

Reply:
@mbabker Do you want to make another PR replacing this one?